WalletWasabi/Services/WalletService.cs

```diff
@@ -622,7 +622,28 @@ private async Task<bool> ProcessTransactionAsync(SmartTransaction tx)
 				}
 			}
 
-			List<SmartCoin> spentOwnCoins = null; 
+			var isLikelyCoinJoinOutput = false;
+			bool hasEqualOutputs = tx.Transaction.GetIndistinguishableOutputs(includeSingle: false).FirstOrDefault() != default;
+			if (hasEqualOutputs)
+			{
+				var receiveKeys = KeyManager.GetKeys(x => tx.Transaction.Outputs.Any(y => y.ScriptPubKey == x.P2wpkhScript));
+				bool allReceivedInternal = receiveKeys.All(x => x.IsInternal);
+				if (allReceivedInternal)
+				{
+					// It is likely a coinjoin if the diff between receive and sent amount is small and have at least 2 equal outputs.
+					Money spentAmount = Coins.Where(x => tx.Transaction.Inputs.Any(y => y.PrevOut.Hash == x.TransactionId && y.PrevOut.N == x.Index)).Sum(x => x.Amount);
+					Money receivedAmount = tx.Transaction.Outputs.Where(x => receiveKeys.Any(y => y.P2wpkhScript == x.ScriptPubKey)).Sum(x => x.Value);
+					bool receivedAlmostAsMuchAsSpent = spentAmount.Almost(receivedAmount, Money.Coins(0.005m));
+
+					if (receivedAlmostAsMuchAsSpent)
+					{
+						isLikelyCoinJoinOutput = true;
+					}
```

Comment:
nit: Could just do
```
isLikelyCoinJoinOutput = spentAmount.Almost(receivedAmount, Money.Coins(0.005m));
```

Reply:
I think the original one is easier to read.
@benthecarman  While your code is correct in this particular case, consider this scenario:
Let's say the `isLikelyCoinJoinOutput` variable is `true` before the `if (receivedAlmostAsMuchAsSpent)` statement, using `isLikelyCoinJoinOutput = spentAmount.Almost(receivedAmount, Money.Coins(0.005m));` would be wrong.

Reply:
Yes but we are setting `isLikelyCoinJoinOutput` to false beforehand here.

Reply:
That is true, and that is why I said your code is correct in this case.
I was just trying to point out that when we set a bool variable to true/false in the if part and we set it to the opposite in the else part that kind of refactoring makes perfect sense (if it is readable of course), but one might make a mistake when there is only the if part :).